ampd/src/handlers/evm_verify_msg.rs

```diff
@@ -352,8 +353,78 @@ mod tests {
 
     #[test]
     fn should_deserialize_correct_event() {
-        let event: Result<PollStartedEvent, events::Error> = (&get_poll_started_event()).try_into();
+        let event: Event = get_event(
+            get_poll_started_event(participants(5, None), 100),
+            &TMAddress::random(PREFIX),
+        );
+        let event: Result<PollStartedEvent, events::Error> = event.try_into();
 
         assert!(event.is_ok());
     }
+
+    #[async_test]
+    async fn should_skip_expired_poll() {
+        let mut rpc_client = MockEthereumClient::new();
+        // mock the rpc client as erroring. If the handler successfully ignores the poll, we won't hit this
+        rpc_client.expect_finalized_block().returning(|| {
+            Err(Report::from(ProviderError::CustomError(
+                "failed to get finalized block".to_string(),
+            )))
+        });
```

Comment:
`handle` just returns `Ok(())` whether the poll was skipped or not. Forcing the rpc client to return an error means `handle` will return an error if the poll was not skipped, and otherwise will return `Ok(())`

Reply:
but if the expectation is not set at all, the mock would fail on its own when the method is called.

Reply:
yes, but in that situation it will panic instead of error, which is tricky to handle